streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -680,7 +693,7 @@ private void removeStandbyTasks() {
         }
     }
 
-    private class StreamsMetricsImpl implements StreamsMetrics {
+    protected class StreamsMetricsImpl implements StreamsMetrics {
```

Comment:
Since there are now two implementations of `StreamsMetrics`, is it confusing to have them both named `StreamsMetricsImpl`? This could be `ThreadStreamsMetrics` and the other could be `ProcessorNodeStreamsMetrics`.